lib/sda/host_msg.go

```diff
@@ -77,13 +86,24 @@ func (t *Token) Id() TokenID {
 	defer tokenMutex.Unlock()
 	if t.cacheId == TokenID(uuid.Nil) {
 		url := network.NamespaceURL + "token/" + t.EntityListID.String() +
-			t.RoundID.String() + t.ProtoID.String() + t.TreeID.String() +
+			t.RoundID.String() + t.ServiceID.String() + t.ProtoID.String() + t.TreeID.String() +
 			t.TreeNodeID.String()
 		t.cacheId = TokenID(uuid.NewV5(uuid.NamespaceURL, url))
 	}
 	return t.cacheId
 }
 
+func (t *Token) Clone() *Token {
+	t2 := Token{}
```

Comment:
Remember the > 2-days bug in the shamir library, where we even ask for Bryan some help ? 
It was because I also thought that doing the indirection would make a copy but .. Not ! ;)